From: Hannes Reinecke <hare@suse.de>
Date: Tue, 3 Mar 2015 13:32:02 +0100
Subject: scsi_dh_alua: recheck RTPG in regular intervals
References: bsc#895814
Patch-Mainline: submitted to linux-scsi

We cannot rely on the sense code to inform us about any
state changes, so we should use a regular interval for
rechecking RTPG. Otherwise we might incur a long delay
before we notice a state change.

Signed-off-by: Hannes Reinecke <hare@suse.de>
---
 drivers/scsi/device_handler/scsi_dh_alua.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/scsi/device_handler/scsi_dh_alua.c b/drivers/scsi/device_handler/scsi_dh_alua.c
index 46f1c0c..76bb081 100644
--- a/drivers/scsi/device_handler/scsi_dh_alua.c
+++ b/drivers/scsi/device_handler/scsi_dh_alua.c
@@ -767,7 +767,7 @@ static int alua_rtpg(struct scsi_device *sdev, struct alua_port_group *pg)
 	case TPGS_STATE_TRANSITIONING:
 		if (time_before(jiffies, pg->expiry)) {
 			/* State transition, retry */
-			pg->interval += 2;
+			pg->interval = 2;
 			err = SCSI_DH_RETRY;
 		} else {
 			/* Transitioning time exceeded, set port to standby */
-- 
1.8.5.2

